### PR TITLE
fix(dependencies): handle --no-install flag properly

### PR DESCRIPTION
**Pull request by emmanuelchucks**
This allows to run this command and skip all prompts. Useful when called from a script.

```bash
pnpm create hono my-app --template x-basic --no-install --pm pnpm
```

### CODE REVIEW

**Comment by yusukebe** [comment]
@emmanuelchucks I'll check it later!

File: src/hooks/dependencies.ts [hunk 1]
```diff
@@ -30,7 +30,13 @@ const registerInstallationHook = (
 
     if (installArg) {
```
Comment (yusukebe) [inline src/hooks/dependencies.ts:31]: I think we can write the logic with this code:

```ts
if (typeof installArg === 'boolean') {
  installDeps = installArg
} else {
  installDeps = await confirm({
    message: 'Do you want to install project dependencies?',
    default: true,
  })
}
```

**Review by yusukebe — COMMENTED** [review]
The below type definition may be wrong:

https://github.com/honojs/create-hono/pull/39/files#diff-53fa883b38520f2227061a3f1a05361640f41438c36170ae58efe4b340ccdb3aR23

It should be `boolean | undefined`.

**Comment by yusukebe** [comment]
Hi @emmanuelchucks 

Thank you for the PR. I've left comments.

File: src/hooks/dependencies.ts (hunk 1 above)
Comment (emmanuelchucks) [reply src/hooks/dependencies.ts]: You're right. I've made the change

**Comment by yusukebe** [comment]
@emmanuelchucks 

Thanks! Let's ship it.

**Pull request merged**